interview-questions.md

```diff
@@ -58,11 +69,16 @@ INSTRUCTOR EXAMPLE: What is a conditional statement?
 Research and define the following terms to the best of your ability.
 
 1. Higher Order Functions:
+- Higher-Order functions take in and act on another function and they can return a function. Their parameters will act upon arrays. They can take can take a specified array and return a new array with modified outputs based on what you asked it to do. Higher-Order functions are sort of a way to shorten code because it can take on functions to produce a modified efficient shorthand version of them. These functions preform by understanding the value of given functions and values of arrays in order to create the expected outputs.
 
 2. Jest:
+- Jest most definitely is comparable to the scientific method, in my opinion. It will be the best way to avoid missing the essentials of your code. It makes you think about what you want out of your code. For example, the (describe) method lists the name of the function kind of giving you the observation of what's to come. The (test) or (it) is basically like your hypothesis of the function and explains what it is doing. Lastly the (expect) is similar to the prediction because it is supposed to call the function being tested and what is expected. Then after doing the (describe) method, when you run (yarn jest) in terminal, it is testing your prediction. Which will give you a lot of information on what went wrong and where you need to go back to the drawing board in order to optimize the code.
```

Comment:
Incredible answer!  I wish I had seen this before teaching Jest this week, I would have added it into my lecture notes, great analogy!!